src/python/zquantum/optimizers/basin_hopping.py

```diff
@@ -0,0 +1,96 @@
+from zquantum.core.interfaces.optimizer import Optimizer, optimization_result
+from zquantum.core.history.recorder import recorder
+from typing import Callable, Dict, Union
+import scipy.optimize
+
+
+class BasinHoppingOptimizer(Optimizer):
+    def __init__(
+        self,
+        keep_value_history: bool = False,
```

Comment:
The `Optimizer` interface says that all of these should be included in the `options` dictionary.
I don't say it's a good design pattern, but now it's inconsistent across optimizers.
WDYT?